### ISSUE TITLE
no-use-before-define: Gives out wrong warnings

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#1629]
Under this scenario:

```.js
function abc(){
    console.log("I am abc function");
    xyz();
}

function xyz(){
    console.log("I am xyz function");
}

abc();
```

It reports an error message saying:
```
3	4	Warning	xyz was used before it was defined	 (no-use-before-define)
```

But I think this should not be a warning/error?

**Comment by gyandeeps** [comment]
Had to use the parameter `nofunc`. So not an issue.
